test/integration/test_forward.rb

```diff
@@ -644,4 +656,69 @@ def test_forward_local_unix_socket_to_remote_port
       end
     end
   end
+
+  def test_forward_local_unix_socket_to_remote_socket
+    setup_ssh_env do
+      start_sshd do |_pid, port|
+        session = Timeout.timeout(4) do
```

Comment:
That's what I had initially. Unfortunately, it doesn't work. I didn't look too much into why, but either sshd gets grumpy when you open a socket and close it again without any additional communication or _port-available_ doesn't actually mean sshd is ready to open a session.

At any rate, opening a session is the only reliable way I found.